pilot/pkg/model/sidecar.go

```diff
@@ -220,11 +218,25 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 	out.namespaceDependencies = make(map[string]struct{})
 	for _, listener := range out.EgressListeners {
 		for _, s := range listener.services {
-			// TODO: port merging when each listener generates a partial service
 			if _, found := servicesAdded[string(s.Hostname)]; !found {
-				servicesAdded[string(s.Hostname)] = struct{}{}
+				servicesAdded[string(s.Hostname)] = s
 				out.services = append(out.services, s)
 				out.namespaceDependencies[s.Attributes.Namespace] = struct{}{}
+			} else if s.Ports != nil && len(s.Ports) > 0 {
+				// merge the ports to service when each listener generates partial service
+				os := servicesAdded[string(s.Hostname)]
+				for _, p := range s.Ports {
```

Comment:
@rshriram This is the fix I have put. Earlier, it was duplicate ports while merging.

Reply:
How is duplicate ports possible in an egress listener though?

Reply:
good point. I did see that in debug when services are merged by listener, each listener has the same service with same port - Now that you mentioned this point, seems there is a problem. Let me debug again.

Reply:
If you are wondering about test case, I incorrectly changed the test case earlier - which I corrected in this PR as well